cartographer_grpc/framework/rpc.h

```diff
@@ -36,12 +36,20 @@ namespace cartographer_grpc {
 namespace framework {
 
 class Service;
+// TODO(cschuet): Add a unittest that tests the logic of this class.
 class Rpc {
  public:
   struct RpcEvent;
   using EventQueue = cartographer::common::BlockingQueue<RpcEvent*>;
   using WeakPtrFactory = std::function<std::weak_ptr<Rpc>(Rpc*)>;
-  enum class Event { NEW_CONNECTION = 0, READ, WRITE, FINISH, DONE };
+  enum class Event {
+    NEW_CONNECTION = 0,
+    READ,
+    WRITE_NEEDED,
```

Comment:
So this new event is needed to have an additional pending flag?

Reply:
Rather the other way around: We don't really care about the pending flag, but we need this event to notify our handler thread that there is something to send in the outbox.